examples/demo3/src/backend/server/demo3_service.go

```diff
@@ -0,0 +1,1425 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/InterDigitalInc/AdvantEDGE/example/demo3/src/util"
+	ams "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-ams-client"
+	asc "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-app-support-client"
+	log "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-logger"
+	sbx "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-sandbox-ctrl-client"
+	smc "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-service-mgmt-client"
+	"github.com/gorilla/mux"
+)
+
+var mutex sync.Mutex
+var intervalTicker *time.Ticker
+var done chan bool
+
+// App-enablement client
+var srvMgmtClient *smc.APIClient
+var srvMgmtClientPath string
+var appSupportClient *asc.APIClient
+var appSupportClientPath string
+
+// Sandbox controller client
+var sandBoxClient *sbx.APIClient
+var sbxCtrlUrl string = "http://meep-sandbox-ctrl"
+
+// Ams client & context transfer payload
+var amsClient *ams.APIClient
+var amsResourceId string
+var amsTargetId string
+var amsServiceName string
+
+var svcSubscriptionSent bool
+var appTerminationSent bool
+var serviceRegistered bool
+var amsSubscriptionSent bool
+var amsServiceCreated bool
+var terminated bool
+var terminateNotification bool
+var appEnablementEnabled bool
+
+// Config fields
+var mecUrl string
+var localPort string
+var localUrl string
+var environment string
+var callBackUrl string
+
+// App service & discovered service
+var mecServicesMap = make(map[string]string)
+var instanceName string
+var serviceCategory string = "demo3"
+var mep string
+var serviceAppVersion string = "v2.1.1"
+var scopeOfLocality string = defaultScopeOfLocality
+var consumedLocalOnly bool = defaultConsumedLocalOnly
+
+const defaultScopeOfLocality = "MEC_SYSTEM"
+const defaultConsumedLocalOnly = true
+
+// Demo models
+var demoAppInfo ApplicationInstance
+var appActivityLogs []string
+var subscriptions ApplicationInstanceSubscriptions
+var appEnablementServiceId string
+var amsSubscriptionId string
+var terminationSubscriptionId string
+
+// Ams state
+var trackDevices []string                      // Devices currently using instance with incremental state
+var terminalDevices map[string]string          // All devices registered under ams and their message info
+var terminalDeviceState = make(map[string]int) // All devices registered under ams and their state
+var orderedAmsAdded = []string{}
+
+// Initiaze ticker to increment terminal device state every second
+func startTicker() {
+	intervalTicker = time.NewTicker(1 * time.Second)
+	go func() {
+		for range intervalTicker.C {
+			// Increment terminal device state by 1
+			for _, device := range trackDevices {
```

Comment:
Playing with global variables here... should be protected by the Mutex

Reply:
after making changes, the demo3 ams state cannot be updated after adding a terminal device